Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/SubtreeTagging.php

```diff
@@ -146,20 +149,36 @@ private function moveSubtreeTags(ContentStreamId $contentStreamId, NodeAggregate
                   ' . $this->getTableNamePrefix() . '_hierarchyrelation th
                   INNER JOIN ' . $this->getTableNamePrefix() . '_node tn ON tn.relationanchorpoint = th.childnodeanchor
                 WHERE
-                  tn.nodeaggregateid = :nodeAggregateId
+                  tn.nodeaggregateid = :newParentNodeAggregateId
                   AND th.contentstreamid = :contentStreamId
                   AND th.dimensionspacepointhash = :dimensionSpacePointHash
                 UNION
-                SELECT JSON_MERGE(cte.subtreetagsToInherit, JSON_KEYS(JSON_MERGE_PATCH(\'{}\', dh.subtreetags))) subtreeTagsToInherit, dh.childnodeanchor
+                SELECT
+                    JSON_MERGE_PRESERVE(
+                        cte.subtreeTagsToInherit,
+                        JSON_KEYS(JSON_MERGE_PATCH(
+                            \'{}\',
+                            dh.subtreetags
+                        ))
+                    ) subtreeTagsToInherit,
+                    dh.childnodeanchor
                 FROM
                   cte
-                  JOIN ' . $this->getTableNamePrefix() . '_hierarchyrelation dh ON dh.parentnodeanchor = cte.childnodeanchor
+                JOIN ' . $this->getTableNamePrefix() . '_hierarchyrelation dh
+                    ON
+                        dh.parentnodeanchor = cte.childnodeanchor
+                    WHERE
+                        dh.contentstreamid = :contentStreamId
+                        AND dh.dimensionspacepointhash = :dimensionSpacePointHash
```

Comment:
Just wondering:
```sql
                    ON
                        dh.parentnodeanchor = cte.childnodeanchor
                        AND dh.contentstreamid = :contentStreamId
                        AND dh.dimensionspacepointhash = :dimensionSpacePointHash
```
would be the same – do you know if there is any (performance?) impact on one vs the other? I doubt there is

Reply:
agreed; I don't know about any performance impact, so I adjusted this for better readability